rust/gui-client/src/welcome.ts

```diff
@@ -1,9 +1,47 @@
 import { invoke } from "@tauri-apps/api/core";
+import { listen } from '@tauri-apps/api/event';
+
+interface Session {
+  account_slug: string,
+  actor_name: string,
+}
 
 const signInBtn = <HTMLButtonElement>document.getElementById("sign-in");
+const signOutBtn = <HTMLButtonElement>document.getElementById("sign-out");
+
+const signedInDiv = <HTMLDivElement>document.getElementById("signed-in");
+const signedOutDiv = <HTMLDivElement>document.getElementById("signed-out");
+
+const accountSlugSpan = <HTMLSpanElement>document.getElementById("account_slug");
+const actorNameSpan = <HTMLSpanElement>document.getElementById("actor_name");
+
+// Initial state is to assume we are signed out.
+signedOutDiv.style.display = "block";
+signedInDiv.style.display = "none";
 
 signInBtn.addEventListener("click", async (_e) => {
   console.log("Signing in...");
 
   await invoke("sign_in");
 });
+signOutBtn.addEventListener("click", async (_e) => {
+  console.log("Signing in...");
+
+  await invoke("sign_out");
+});
+
+listen<Session>('signed_in', (e) => {
+  let session = e.payload;
+
+  accountSlugSpan.textContent = session.account_slug;
+  actorNameSpan.textContent = session.actor_name;
+  signedOutDiv.style.display = "none";
+  signedInDiv.style.display = "block";
+})
+
+listen<void>('signed_out', (_e) => {
+  accountSlugSpan.textContent = "";
+  actorNameSpan.textContent = "";
+  signedOutDiv.style.display = "block";
+  signedInDiv.style.display = "none";
+})
```

Comment:
I am using something new here. Tauri has the ability to send "events" to the frontend and we can listen on those events and update the GUI as a result. This allows us to nicely decouple things. If we get signed out for whatever reason and this window happens to be open, it will automatically update.

I am gonna do the same thing for the settings once I revamp those.

Reply:
This is how we do it on Apple and Android. The tunnel can run in whatever state it's in. When a GUI connects, it automatically gets updated with the proper state appropriately.

Reply:
Main difference that I want to point out here is that the communication is one-way. We never request data, it gets pushed via an event.

Reply:
Also, this is between WebView and GUI client, the tunnel process is not at play here.